#🧠 O que são?

* Testes paramétricos (como t, F, ANOVA) assumem que os dados seguem uma distribuição conhecida, geralmente a normal.

* Testes não-paramétricos são alternativas robustas que não exigem essa suposição.

# 🏛️ Motivação

* Dados assimétricos,
* Presença de outliers,
* Distribuições desconhecidas,
* Variáveis ordinais (ex: grau de dor: leve, moderada, intensa),
* Amostras pequenas sem garantia de normalidade.

**Exemplo salário**

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import ttest_ind, mannwhitneyu

In [3]:
# Simular dois grupos de salários
np.random.seed(42)
n = 100

In [4]:
# Grupo A: distribuição de salários com assimetria e outliers (mediana = 3000)
salarios_a = np.concatenate([np.random.normal(3000, 500, n), np.array([20000, 25000])])  # outliers

# Grupo B: semelhante, mas com média um pouco maior
salarios_b = np.concatenate([np.random.normal(3300, 500, n), np.array([21000, 26000])])

# Organizar DataFrame
df_salarios = pd.DataFrame({
    "salario": np.concatenate([salarios_a, salarios_b]),
    "grupo": ["A"] * len(salarios_a) + ["B"] * len(salarios_b)
})

In [5]:
df_salarios

salario grupo
0     3248.357077     A
1     2930.867849     A
2     3323.844269     A
3     3761.514928     A
4     2882.923313     A
..            ...   ...
199   3376.862553     B
200   3329.104359     B
201   2728.514851     B
202  21000.000000     B
203  26000.000000     B

[204 rows x 2 columns]

In [6]:
# Calcular estatísticas
media_a, mediana_a = np.mean(salarios_a), np.median(salarios_a)
media_b, mediana_b = np.mean(salarios_b), np.median(salarios_b)

In [11]:
# Mostrar tabela de resultados
comparacao = pd.DataFrame({
    "Grupo": ["A", "B"],
    "Média": [media_a, media_b],
    "Mediana": [mediana_a, mediana_b]
}).round(2)

In [12]:
comparacao

Grupo    Média  Mediana
0     A  3331.45  2953.09
1     B  3707.01  3363.34

In [10]:
# Aplicar testes
t_stat, t_p = ttest_ind(salarios_a, salarios_b)
mw_stat, mw_p = mannwhitneyu(salarios_a, salarios_b)

In [13]:
resultados_teste = pd.DataFrame({
    "Teste": ["t-test", "Mann-Whitney U"],
    "Estatística": [t_stat, mw_stat],
    "p-valor": [t_p, mw_p]
}).round(4)

In [14]:
resultados_teste

Teste  Estatística  p-valor
0          t-test      -0.9481   0.3442
1  Mann-Whitney U    3195.0000   0.0000

✅ t-test (paramétrico): Comparação de médias assumindo distribuições normais.
* Resultado: p = 0.3442 → não rejeita H₀.
* Conclusão: segundo o teste t, as médias dos salários não são diferentes.
* Isso acontece porque: O teste t é sensível à média, e os outliers do Grupo B aumentaram a média, mas com alta variabilidade, o teste não detecta diferença estatística.

✅ Mann-Whitney U (não paramétrico): Comparação das distribuições com base em ranks, mais robusto a outliers.
* Resultado: p < 0.001 → rejeita H₀.
* Conclusão: segundo o teste Mann-Whitney, há uma diferença significativa entre os grupos.
* Isso acontece porque: O teste avalia a mediana e distribuição geral, e o Grupo B, mesmo com mediana próxima, apresenta uma distribuição assimétrica e dispersa, que afeta a comparação de rankings.

# 🔄 Comparação com testes paramétricos

| Aspecto               | Paramétrico               | Não Paramétrico                        |
| --------------------- | ------------------------- | -------------------------------------- |
| Supõe normalidade     | Sim                       | Não                                    |
| Mede                  | Média                     | Mediana ou ranking                     |
| Resistente a outliers | Não                       | Sim                                    |
| Uso comum             | Dados contínuos e normais | Ordinais, assimétricos, pequenos       |
| Exemplo típico        | t-test, ANOVA             | Mann-Whitney, Wilcoxon, Kruskal-Wallis |


# 🧪 Testes equivalentes

| Finalidade                   | Paramétrico     | Não Paramétrico            |
| ---------------------------- | --------------- | -------------------------- |
| 2 grupos independentes       | t-test          | Mann-Whitney U             |
| 2 grupos dependentes         | t pareado       | Wilcoxon                   |
| 3+ grupos independentes      | ANOVA           | Kruskal-Wallis             |
| 3+ grupos dependentes        | ANOVA Repetidas | Friedman                   |
| Proporções (binário pareado) | Qui-quadrado    | Teste de sinais / Binomial |


# 🟢 Vantagens dos Testes Não-Paramétricos

* Avaliam a mediana → mais resistente a assimetrias.
* Funcionam com dados ordinais.
* São úteis quando: A amostra é pequena, A distribuição é desconhecida ou distorcida, Há outliers legítimos que você não quer remover.

# 📌 Cuidados

Testes não-paramétricos têm menor poder estatístico quando os dados são normais.

# 💡 Quando usar?

Use um teste não paramétrico quando:

* Você não pode garantir a normalidade,
* Os dados são ordinais,
* Há outliers que refletem a realidade do fenômeno estudado,
* A amostra é pequena.